src/mainwindow.cpp

```diff
@@ -53,7 +53,7 @@
 #include <QStateMachine>
 #include <QTimer>
 #ifdef Q_OS_LINUX
-#include <QX11Info>
+#include <QtX11Extras/QX11Info>
```

Comment:
The linkage and dependencies are valid, but I don't think that you need to change the header, it should be accessible through just `QX11Info`.

Reply:
It should be, but it isn't. I can't understand why, on Qt documentation I only find mentions of "#include <QX11Info>".
The error given is `QX11Info: No such file or directory`.
I am not a developer and this fix is simply the result of an afternoon trying to understand what's wrong.

Reply:
It could be because `Qt5::X11Extras` linked. In CMake `target_link_libraries` also adds necessary include directories. Could you try again with just `#include <QX11Info>`?

Reply:
My process order was the oposite: I first included `<QtX11Extras/QX11Info>`, _then_ cmake complained that QtX11Extras wasn't linked.

Reply:
Interesting...
Do your distro have https://github.com/ksnip/ksnip or https://github.com/flameshot-org/flameshot? They also use `X11Extras` and they use `#include <QX11Info>` in the code.

Reply:
Yes, we have Ksnip 1.10.1 and it builds without patches: https://build.opensuse.org/package/show/openSUSE:Factory/ksnip

I have since found though that another distribution, ALT Linux, made a similar substitution to the one I acted: https://packages.altlinux.org/en/sisyphus/srpms/crow-translate/specfiles/#line-100
They added this patch when they updated to 2.11.0.

Anyway, in the openSUSE mainline package version of crow-translate, albeit at version 2.10.0, c++ seems to correctly include `/usr/include/qt5/QtX11Extras`, which should make it possible to only specify `#include <QX11Info>`…

Reply:
Strange, I didn't do any related changes since 2.10.0.
Maybe something changed in Qt since then?

I'm fine with using `QtX11Extras/QX11Info` for better compatibility, but I also curios why the canonical include of `QX11Info` doesn't work.

Reply:
> Strange, I didn't do any related changes since 2.10.0.
Sorry, you didn't see my edit. ALT updated to 2.11.0 from 2.10.4, so the something strange is exactly the inclusion of QX11Info. It doesn't help

Reply:
I saw it, but what I trying to say is that I had this include for a long time...
It's even present in QHotkey which I used almost since 1.0.

Reply:
I think it's a distro-specific problem.
I will keep the linkage addition since this is 100% correct. But not sure about using `QtX11Extras`.
Feel free to just add this directory to include folder for compilation in your build script.